pkg/service/utils.go

```diff
@@ -0,0 +1,34 @@
+package service
+
+import (
+	"crypto/rand"
+	"crypto/tls"
+
+	"google.golang.org/grpc/credentials"
+)
+
+func loadTLSCredentials(serverCertPath string, serverKeyPath string) (credentials.TransportCredentials, error) {
+	// Load server's certificate and private key
+	creds, err := credentials.NewServerTLSFromFile(serverCertPath, serverKeyPath)
+	if err != nil {
+		return nil, err
+	}
+
+	return creds, nil
+}
+
+func loadTLSCertificate(certPath, keyPath string) (*tls.Certificate, error) {
+	certificate, err := tls.LoadX509KeyPair(certPath,
+		keyPath)
+	if err != nil {
+		return nil, err
+	}
+
+	config := &tls.Config{
+		Certificates: []tls.Certificate{certificate},
+		Rand:         rand.Reader,
+		MinVersion:   tls.VersionTLS12,
+	}
+
+	return &config.Certificates[0], nil
```

Comment:
it looks like the returned certificate is being used to reconstruct the `tls.Config` struct in `HTTPService.tlsListener` it may be simpler to just return the config at this point

Reply:
Good spot, fixed